src/client/dfuse/pil4dfs/hook.c

```diff
@@ -172,77 +179,88 @@ read_map_file(char **buf, const char str_target[], char **pos)
-		return;
+		goto err;
 	}
-	for (i = 0; i < 16; i++) {
-		if (strncmp(pPos + i, ".so", 3) == 0) {
-			pEnd = pPos + i + 3;
-			break;
-		}
+	get_string_pos(pos, &start, &end);
+	if ((end - start + 1) >= PATH_MAX) {
+		printf("Warning: path of path_ld is too long.\n");
+		goto err;
 	}
-	if (pEnd == NULL) {
-		printf("Fail to determine the ending position of libc path.\nQuit\n");
-		exit(1);
+	path_ld = malloc(end - start + 1);
+	if (path_ld == NULL) {
+		printf("Warning: Failed to allocate memory for path_ld.\n");
+		goto err;
 	}
-	for (i = 0; i < 100; i++) {
-		if (strncmp(pPos - i, "  /", 3) == 0) {
-			pStart = pPos - i + 2;
-			break;
-		}
+	memcpy(path_ld, start, end - start);
+	path_ld[end - start] = 0;
+
+	pos = strstr(read_buff_map, "/lib64/libc.so");
+	if (pos == NULL)
+		/* try a different format */
+		pos = strstr(read_buff_map, "/lib64/libc-2");
+	if (pos == NULL) {
+		printf("Warning: Failed to find libc.so.\n");
+		goto err;
 	}
-	if (pStart == NULL) {
-		printf("Fail to determine the starting position of libc path.\nQuit\n");
-		exit(1);
+	get_string_pos(pos, &start, &end);
+	if ((end - start + 1) >= PATH_MAX) {
+		printf("The length of path_libc is too long!\nQuit\n");
+		goto err;
 	}
-
-	if ((long int)(pEnd - pStart) >= PATH_MAX) {
-		printf("The length of path_ld is too long!\nQuit\n");
-		exit(1);
+	path_libc = malloc(end - start + 1);
+	if (path_libc == NULL) {
+		printf("Warning: Failed to allocate memory for path_libc.\n");
+		goto err;
 	}
-	path_ld = malloc((int)(pEnd - pStart) + 1);
-	if (path_ld == NULL) {
-		printf("Failed to allocate memory for path_ld.\nQuit\n");
-		exit(1);
+	_Pragma("GCC diagnostic push")
+	_Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
+	_Pragma("GCC diagnostic ignored \"-Wstringop-overflow\"")
+	_Pragma("GCC diagnostic ignored \"-Wrestrict\"")
+	memcpy(path_libc, start, end - start);
+	path_libc[end - start] = 0;
+	_Pragma("GCC diagnostic pop")
+
+	/* compose path_libpthread with path_libc string */
+	if ((end - start + 1 + 6) >= PATH_MAX) {
+		printf("The length of path_libpthread is too long!\nQuit\n");
+		goto err;
+	}
+	path_libpthread = malloc(end - start + 1 + 6);
+	if (path_libpthread == NULL) {
+		printf("Warning: Failed to allocate memory for path_libpthread.\n");
+		goto err;
 	}
-	memcpy(path_ld, pStart, pEnd - pStart);
-	path_ld[pEnd - pStart] = 0;
 
-	memcpy(lib_ver_str, pPos + 4, pEnd - 3 - (pPos + 4));
-	lib_ver_str[pEnd - 3 - (pPos + 4)] = 0;
-	memcpy(lib_dir_str, pStart, pPos - pStart);
-	lib_dir_str[pPos - pStart] = 0;
+	memcpy(path_libpthread, start, end - start);
+	path_libpthread[end - start + 1 + 5] = 0;
+	pos = strstr(path_libpthread, "libc");
+	memcpy(pos + 3    , "pthread", 7);
+	memcpy(pos + 3 + 7, path_libc + (pos - path_libpthread + 4), end - start + 1 + 6 -
+	       (pos - path_libpthread + 4));
+	if (path_libpthread[end - start + 1 + 4] >= '5' &&
+	    path_libpthread[end - start + 1 + 4] <= '7') {
+		path_libpthread[end - start + 1 + 4] = '0';
+	}
```

Comment:
it's not clear from the code how this differs from the previous version and it's a significant regression in terms of readability.  You can't just have `pos + 7` or similar in the code without some comment about what it's doing any why.  If you can't do this with a combination of `dirname` and `asprintf` then please describe what you're trying to do and I can help you.

It looks like of version is six then you're changing it to zero, if this is indeed the case you need a comment to justify it.